app/gui/src/controller/searcher.rs

```diff
@@ -720,12 +730,64 @@ impl Searcher {
         self.notifier.notify(Notification::NewActionList);
     }
 
+    const AI_QUERY_PREFIX: &'static str = "AI:";
+    const AI_QUERY_ACCEPT_TOKEN: &'static str = "#";
+    const AI_STOP_SEQUENCE: &'static str = "`";
+    const AI_GOAL_PLACEHOLDER: &'static str = "__$$GOAL$$__";
+
+    async fn accept_ai_query(
+        query: String,
+        this: ThisNode,
+        graph: Handle,
+        notifier: notification::Publisher<Notification>,
+    ) -> FallibleResult {
+        let vis_ptr = QualifiedMethodPointer::from_qualified_text(
+            "Standard.Visualization.AI",
+            "Standard.Visualization.AI",
+            "build_ai_prompt",
+        )?;
+        let vis = Visualization::new(this.id, vis_ptr, vec![]);
+        let mut result = graph.attach_visualization(vis.clone()).await?;
+        let next = result.next().await.ok_or(NoAIVisualizationDataReceived)?;
+        let prompt = std::str::from_utf8(&next)?;
+        let prompt_with_goal = prompt.replace(Self::AI_GOAL_PLACEHOLDER, &query);
+        graph.detach_visualization(vis.id).await?;
+        let completion = graph.get_ai_completion(&prompt_with_goal, Self::AI_STOP_SEQUENCE).await?;
+        notifier.publish(Notification::AISuggestionUpdated(completion)).await;
+        Ok(())
+    }
+
+    fn handle_ai_query(&self, query: String) -> FallibleResult {
```

Comment:
I know this is PoC, but docs are needed here for future refactoring/development. Please describe shortly what this function does and why.